syft/pkg/cataloger/config.go

```diff
@@ -24,13 +25,18 @@ func DefaultConfig() Config {
 		Parallelism:                     1,
 		LinuxKernel:                     kernel.DefaultLinuxCatalogerConfig(),
 		Python:                          python.DefaultCatalogerConfig(),
+		Java:                            java.NewJavaCatalogerOpts(),
 		ExcludeBinaryOverlapByOwnership: true,
 	}
 }
 
-func (c Config) Java() java.Config {
+// JavaConfig merges relevant config values from Config to return a java.Config struct.
+// Values like IncludeUnindexedArchives and IncludeIndexedArchives are used across catalogers
+// and are not specific to Java requiring this merge.
+func (c Config) JavaConfig() java.Config {
```

Comment:
very small nit: this rename seems unnecessary, since this is a method on the `Config` object already, it's a bit of a stutter `config.JavaConfig()` rather `config.Java()`

Reply:
Sorry - I think this did have to be renamed in order for the struct to be named consistently.

`Config` already had the fields `Python` and `Golang`. 

I opted to change `Java` from a `function` to the field like the other languages on that struct. This was so that `Java` could use the `CatalogerOpts` pattern like the  `Golang` field used for options that are exclusive to the specific ecosystem.

`config.JavaConfig()` doesn't seem too bad a stutter given the trade off. We still need a function that can grab fields from `Config` which are not mutually exclusive to a single cataloger to generate a `java.Config` (things like `IncludeIndexedArchives`) hence `config.JavaConfig` being born.